### PR TITLE
exclusive-optionals: exclusive optionals

### PR DESCRIPTION
**Pull request by pablomusumeci**

### CODE REVIEW

File: mule-sdk/v/4.0/8.1.5_exclusive_optionals.adoc [hunk 1]
```diff
@@ -1,2 +1,81 @@
 [[_exclusive_optionals]]
 = Exclusive Optionals
+
+== What are exclusive optional parameters?
```
Comment (juandesi) [inline mule-sdk/v/4.0/8.1.5_exclusive_optionals.adoc]: remove this title, leave the introduction for the main title

File: mule-sdk/v/4.0/8.1.5_exclusive_optionals.adoc [hunk 2]
```diff
@@ -1,2 +1,81 @@
...
+While developing a module, we found useful to have a way of establishing the following restriction: 
+
+> "From this set of parameters, only one shall be used at any time"
+
+It happens when you have several ways of configuring the same thing, 
+but you only one of those options should be used.
+
+The SDK provides you a way of enforcing this restriction by using `@ExclusiveOptionals`.
+
+=== @ExclusiveOptionals
```
Comment (juandesi) [inline mule-sdk/v/4.0/8.1.5_exclusive_optionals.adoc]: === -> ==

File: mule-sdk/v/4.0/8.1.5_exclusive_optionals.adoc [hunk 3]
```diff
@@ -1,2 +1,81 @@
...
+
+`@ExclusiveOptionals` can only be used on types used as `@ParameterGroup`.
+For more information about parameter groups, please refer to the <<_parameters, Parameters>> documentation.
+
+Let's start with an example:
+
+.ExclusiveOptionalsExample.java
```
Comment (juandesi) [inline mule-sdk/v/4.0/8.1.5_exclusive_optionals.adoc]: remove doc name

File: mule-sdk/v/4.0/8.1.5_exclusive_optionals.adoc [hunk 4]
```diff
@@ -1,2 +1,81 @@
...
+So far, we presented a restriction that guarantee us that no more than one optional parameter
+can be present at the same time, but what happens if none of them are present? It's 
+still a valid scenario because we are not violating the _no more than one_ statement.
+
+It is not crazy to think that sometimes we may want at least one of them to be present, 
+which in plain english sounds like "Only one of them should be always set, no more, no less".
+
+Doing that is as simple as configuring the annotation with `@ExclusiveOptionals(isOneRequired = true)`.
+
+.OneRequired.java
```
Comment (juandesi) [inline mule-sdk/v/4.0/8.1.5_exclusive_optionals.adoc]: remove title

File: mule-sdk/v/4.0/8.1.5_exclusive_optionals.adoc [hunk 5]
```diff
@@ -1,2 +1,81 @@
...
+
+    @Parameter
+    @Optional
+    private String anotherOptionalParameter;
+}
+----
+
+So, configuring `requiredParameter` but not setting `optionalParameter`
+nor `anotherOptionalParameter` will result in an error.
+
```
Comment (juandesi) [inline mule-sdk/v/4.0/8.1.5_exclusive_optionals.adoc]: maybe a DSL example could be added

File: mule-sdk/v/4.0/8.1.5_exclusive_optionals.adoc [hunk 6]
```diff
@@ -1,2 +1,81 @@
...
+> "From this set of parameters, only one shall be used at any time"
+
+It happens when you have several ways of configuring the same thing, 
+but you only one of those options should be used.
+
+The SDK provides you a way of enforcing this restriction by using `@ExclusiveOptionals`.
+
+=== @ExclusiveOptionals
+
+`@ExclusiveOptionals` can only be used on types used as `@ParameterGroup`.
```
Comment (juandesi) [inline mule-sdk/v/4.0/8.1.5_exclusive_optionals.adoc]: The way that the SDK has to implement this is by using the @ExclusiveOptionals annotation .... (then continues all that you wrote)

File: mule-sdk/v/4.0/8.1.5_exclusive_optionals.adoc (hunk 1 above)
Comment (dukesphere) [reply mule-sdk/v/4.0/8.1.5_exclusive_optionals.adoc]: Awaiting action on Juani's request before merging new content. Thanks!

File: mule-sdk/v/4.0/8.1.5_exclusive_optionals.adoc (hunk 6 above)
Comment (dukesphere) [reply mule-sdk/v/4.0/8.1.5_exclusive_optionals.adoc]: Awaiting action on Juani's request before merging new content. Thanks!
